From c0eb072563accdb30c468c89010e824004187cf9 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:00:18 -0400
Subject: [ethernet] mlx4: Load the IB driver when the device supports IBoE

Message-id: <fed0fecb665480cac2769e4446f04bbf78828fb8.1410425016.git.avadai@redhat.com>
Patchwork-id: 93755
O-Subject: [Patch RHEL7.1 V1 082/197] net/mlx4_core: Load the IB driver when the device supports IBoE
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Or Gerlitz <ogerlitz@mellanox.com>

When checking what protocol drivers to load, the IB driver should be
requested also over Ethernet ports, if the device supports IBoE (RoCE).

Fixes: b046ffe 'net/mlx4_core: Load higher level modules according to ports type'
Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 7855bff42ea9938a0853321256f4c8ce3628aa73)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/main.c b/drivers/net/ethernet/mellanox/mlx4/main.c
index 612b823..1dbee38 100644
--- a/drivers/net/ethernet/mellanox/mlx4/main.c
+++ b/drivers/net/ethernet/mellanox/mlx4/main.c
@@ -750,7 +750,7 @@ static void mlx4_request_modules(struct mlx4_dev *dev)
    has_eth_port = true;
  }
 
- if (has_ib_port)
+ if (has_ib_port || (dev->caps.flags & MLX4_DEV_CAP_FLAG_IBOE))
   request_module_nowait(IB_DRV_NAME);
  if (has_eth_port)
   request_module_nowait(EN_DRV_NAME);
-- 
1.7.1